pandablocks/utils.py

```diff
@@ -56,18 +61,23 @@ def words_to_table(
         if field_info.subtype == "int":
             # First convert from 2's complement to offset, then add in offset.
             packing_value = (value ^ (1 << (bit_length - 1))) + (-1 << (bit_length - 1))
-        elif field_info.labels:
+        elif convert_enum_indices and field_info.labels:
             packing_value = [field_info.labels[x] for x in value]
         else:
-            packing_value = value
+            if bit_length <= 8:
+                packing_value = value.astype(np.uint8)
+            elif bit_length <= 16:
+                packing_value = value.astype(np.uint16)
+            else:
+                packing_value = value.astype(np.uint32)
```

Comment:
What's the logic behind these changes? These aren't obviously anything to do with enum fields or conversions? This also appears to be reverting changes that were previously made - this form of conversion existed in commit 39af918 but was later replaced.

Reply:
Ah I changed it because the explicit packing as uint was unnecessary I'll change it back.